File: suites/01_algorithms/comparisons/edit_distance.c
#include <stdio.h>

static long long edit_distance_cost(long long left, long long right) {
    long long total = 0;
    for (long long row = 0; row < left; ++row) {
        for (long long col = 0; col < right; ++col) {
            long long mismatch = ((row * 17) + (col * 13)) % 11;
            if (mismatch < 2) {
                total += 1;
            } else {
                total += (row + col) % 3;
            }
        }
    }
    return total;
}

int main(void) {
    printf("%lld\n", edit_distance_cost(2800, 2800));
    return 0;
}

File: suites/01_algorithms/comparisons/edit_distance.rs
fn edit_distance_cost(left: i64, right: i64) -> i64 {
    let mut total: i64 = 0;
    for row in 0..left {
        for col in 0..right {
            let mismatch = ((row * 17) + (col * 13)) % 11;
            if mismatch < 2 {
                total += 1;
            } else {
                total += (row + col) % 3;
            }
        }
    }
    total
}

fn main() {
    println!("{}", edit_distance_cost(2800, 2800));
}
